src/main/java/util/Utility.java

```diff
@@ -299,4 +299,26 @@ public static boolean startsWithIgnoreCase(String source, String query) {
         return source.toLowerCase().startsWith(query.toLowerCase());
     }
 
+    public static String generateName(String desiredName, List<String> existingNames) {
+        String availableName = desiredName;
+
+        if (existingNames.contains(desiredName)) {
+            List<String> existingSuffixes = existingNames.stream()
+                    .filter(existing -> existing.startsWith(desiredName)
+                            && !existing.equalsIgnoreCase(desiredName))
+                    .map(existing ->
+                            existing.substring(existing.indexOf(desiredName, 0) + desiredName.length()))
+                    .collect(Collectors.toList());
+
+            int index = 1;
+
+            while (existingSuffixes.contains(Integer.toString(index))) {
+                index++;
+            }
+
+            availableName = desiredName + Integer.toString(index);
+        }
+
+        return availableName;
```

Comment:
Oh, makes sense. Didn't realize that because I put the second statement afterwards due to problems.